src/OrbitGl/TimelineUi.cpp

```diff
@@ -59,15 +59,22 @@ void TimelineUi::RenderLabels(Batcher& batcher, TextRenderer& text_renderer,
     }
     float world_x = timeline_info_interface_->GetWorldFromUs(
         tick_ns / static_cast<double>(kNanosecondsPerMicrosecond));
-    if (world_x > previous_label_end_x) {
+    if (world_x > previous_label_end_x + kLabelMarginRight + kPixelMargin) {
       Vec2 pos, size;
-      float label_bottom_y = GetPos()[1] + GetHeight() - kLabelMarginBottom;
+      float label_bottom_y =
+          GetPos()[1] + (GetHeight() + layout_->GetFontSize()) / 2 - kLabelMarginBottom;
       text_renderer.AddText(label.c_str(), world_x + kLabelMarginLeft, label_bottom_y,
                             GlCanvas::kZValueTimeBar,
                             {layout_->GetFontSize(), Color(255, 255, 255, 255), -1.f,
                              TextRenderer::HAlign::Left, TextRenderer::VAlign::Bottom},
                             &pos, &size);
       previous_label_end_x = pos[0] + size[0];
+
+      // Box between labels and ticks, so they don't overlap.
```

Comment:
Will this work? Text is drawn after boxes, and text and box are on the same layer, so the label should be drawn on top of the box. Similar to what I stated in the other PR, we really should try to not "hide" elements behind the background color because it'll just cause issues later.

Reply:
This is currently working, I'm only making it a bit bigger. But yes, we are hiding elements behind background color. Again, a quick discussion could be useful.

Reply:
Ah, I misunderstood - this box is drawn as a text background, I misunderstood the comment. Maybe rephrase the comment to say "Box behind the label to hide the ticks behind it" or something? But now I get it, makes sense.